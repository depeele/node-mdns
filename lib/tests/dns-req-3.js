var Util    = require('util'),
    Utils   = require('../utils.js'),
    Message = require('../message.js'),

    data    = [
        0x00, 0x00, 0x00, 0x00, 0x00, 0x02, 0x00, 0x00,
        0x00, 0x00, 0x00, 0x00, 0x05, 0x5f, 0x68, 0x74,
        0x74, 0x70, 0x04, 0x5f, 0x74, 0x63, 0x70, 0x00,
        0x00, 0x01, 0x00, 0x01, 0xc0, 0x0c, 0x00, 0x1c,
        0x00, 0x01
    ];

/*****************************************************************************
 * Unpack test
 *
 */
console.log("--------------------------------------------------------------");
var rBuf    = new Buffer( data ),
    message = new Message( rBuf );

console.log("Unpack mesage from %d bytes of raw data:", message.consumed);
console.log("message:\n%s", Util.inspect(message, false, 20));

/*****************************************************************************
 * Pack test
 *
 */
console.log("--------------------------------------------------------------");
var wBuf    = new Buffer( 4096 );

message.pack(wBuf);

console.log("Packed message (%d bytes):\n%s",
            message.produced, Utils.buf2hex(wBuf, {
                                            length:     message.produced,
                                            octetsPer:  16
                             }));

if (message.produced !== data.length)
{
    console.log("*** Pack error: length %d != expected %d",
                message.produced, data.length);
    process.exit(-1);
}

var diff    = 0;
for (var idex = 0, len = data.length; idex < len; idex++)
{
    if (wBuf[idex] !== data[idex])
    {
        diff++;
        console.log("*** Pack error: @ %s: %s != expected %s",
                    Utils.int2hex(idex, 2),
                    Utils.int2hex(wBuf[idex]),
                    Utils.int2hex(data[idex]));

        if (diff > 8)   { break; }
    }
}

if (diff === 0)
{
    console.log(">>> packed buffer matches original source");
}
