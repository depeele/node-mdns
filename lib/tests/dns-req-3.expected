--------------------------------------------------------------
Unpack mesage from 34 bytes of raw data:
message:
{ header: 
   { id: 0,
     qr: 0,
     opcode: 0,
     aa: 0,
     tc: 0,
     rd: 0,
     ra: 0,
     z: 0,
     ad: 0,
     cd: 0,
     rcode: 0,
     qdCount: 2,
     anCount: 0,
     nsCount: 0,
     arCount: 0,
     consumed: 12 },
  question: 
   [ { qname: '_http._tcp.', qtype: 1, qclass: 1, consumed: 16 },
     { qname: '_http._tcp.', qtype: 28, qclass: 1, consumed: 6 } ],
  rr: [],
  consumed: 34 }
--------------------------------------------------------------
Header: flags[ 0x00 ]: qr[ 0 ], opcode[ 0 ], aa[ 0 ], tc[ 0 ], rd[ 0 ], ra[ 0 ], z[ 0 ], ad[ 0 ], cd[ 0 ], rcode[ 0 ]
Packed message (34 bytes):
 0x0000: 00 00 00 00 00 02 00 00 00 00 00 00 05 5f 68 74 
 0x0010: 74 70 04 5f 74 63 70 00 00 01 00 01 c0 0c 00 1c 
 0x0020: 00 01
>>> packed buffer matches original source
